sycl/test/basic_tests/handler/handler_generic_lambda_interface.cpp

```diff
@@ -53,10 +53,6 @@ int main() {
   test_parallel_for_work_group<class Group2Name, sycl::group<2>>(sycl::range<2>{1, 1});
   test_parallel_for_work_group<class Group3Name, sycl::group<3>>(sycl::range<3>{1, 1, 1});
 
-#ifndef PREVIEW_BREAKING_CHANGES
-  // For backwards compatibility, we still support some invalid variants of
-  // kernel lambdas. These test cases should be removed once we make conformant
-  // mode the default.
```

Comment:
I'm confused. The comment here says that we still support invalid variants and hides the tests behind the `#ifndef` .

But here you are removing the guards and just running the tests.  So these variants are _not_ invalid?

Reply:
Strictly speaking, they are invalid according to the current wording of SYCL 2020 spec. However, we have received feedback from @gmlueck that in this case we would like to lift the spec restrictions instead of  breaking existing code

Reply:
We actually think this is a bug in the spec, and the spec should allow "auto &".  I have a TODO task to propose a PR that changes the spec.